File: meshlink-proof/ios/ProofApp/ProofBenchmarkTransport.swift
import Foundation

enum ProofBenchmarkTransport {
    case meshLink
    case gattPrototype
    case gattNotifyPrototype

    static func parse(_ rawValue: String?) -> ProofBenchmarkTransport {
        switch rawValue?.lowercased() {
        case "gatt", "gattprototype", "gatt-prototype":
            return .gattPrototype
        case "gatt-notify", "gattnotify", "gattnotifyprototype", "gatt-notify-prototype":
            return .gattNotifyPrototype
        default:
            return .meshLink
        }
    }

    var logLabel: String {
        switch self {
        case .meshLink:
            return "meshlink"
        case .gattPrototype:
            return "gattPrototype"
        case .gattNotifyPrototype:
            return "gattNotifyPrototype"
        }
    }
}

File: meshlink-proof/android/app/src/main/kotlin/ch/trancee/meshlink/proof/android/ProofBenchmarkTransport.kt
package ch.trancee.meshlink.proof.android

import java.util.Locale

internal enum class ProofBenchmarkTransport {
    MeshLink,
    GattPrototype,
    GattNotifyPrototype,
}

internal fun parseProofBenchmarkTransport(rawValue: String?): ProofBenchmarkTransport {
    return when (rawValue?.lowercase(Locale.US)) {
        "gatt", "gattprototype", "gatt-prototype" -> ProofBenchmarkTransport.GattPrototype
        "gatt-notify", "gattnotify", "gattnotifyprototype", "gatt-notify-prototype" ->
            ProofBenchmarkTransport.GattNotifyPrototype
        else -> ProofBenchmarkTransport.MeshLink
    }
}
